subsys/fs/ext2/ext2_ops.c

```diff
@@ -195,7 +194,7 @@ static int ext2_lseek(struct fs_file_t *filp, off_t off, int whence)
 	}
 
 	/* New offset not inside the file. */
-	if (new_off < 0 || new_off > f->f_inode->i_size) {
+	if (new_off > f->f_inode->i_size) {
 		return -EINVAL;
 	}
```

Comment:
I am not sure if it is a proper fix. I would rather change the type of `new_off` to `int32_t` to check if resulting offset is not negative.

Reply:
I changed it to type `off_t`